File: crossecore-java/src/java_/MethodRef.java
package java_;
public interface MethodRef 
 extends ASTNode
{
	
	public static com.crossecore.ocl.QuickSet<MethodRef> allInstances_ = new com.crossecore.ocl.QuickSet<MethodRef>(MethodRef.class);
	
		
	public static com.crossecore.ocl.QuickSet<MethodRef> allInstances(){
		
		com.crossecore.ocl.QuickSet<MethodRef> result = new com.crossecore.ocl.QuickSet<MethodRef>(MethodRef.class);
		result.addAll(MethodRef.allInstances_);
		
		
		return result;
	}
	
	//public static com.crossecore.ocl.QuickSet<MethodRef> allInstances = new com.crossecore.ocl.QuickSet<MethodRef>(MethodRef.class);
	
	
	AbstractMethodDeclaration getMethod();
	void setMethod(AbstractMethodDeclaration value);
	
	TypeAccess getQualifier();
	void setQualifier(TypeAccess value);
	com.crossecore.ocl.OrderedSet<MethodRefParameter> getParameters();
	
}

File: crossecore-typescript/java_/MethodRef.ts
import {OrderedSet} from "ecore/OrderedSet";
import {Set} from "ecore/Set";
import {ASTNode} from "./ASTNode";
import {MethodRefParameter} from "./MethodRefParameter";
import {TypeAccess} from "./TypeAccess";
import {AbstractMethodDeclaration} from "./AbstractMethodDeclaration";

export interface MethodRef
extends ASTNode

{
	
	method:AbstractMethodDeclaration;
	qualifier:TypeAccess;
	parameters: OrderedSet<MethodRefParameter>;
	

}

